src/main/java/com/amazonaws/encryptionsdk/AwsCrypto.java

```diff
@@ -383,8 +474,12 @@ public long estimateCiphertextSize(
 
     /**
      * @see #decryptData(MasterKeyProvider, byte[])
+     *
+     * @deprecated {@link MasterKeyProvider}s have been deprecated in favor of {@link Keyring}s.
+     *              Replaced by {@link #decryptData(AwsCryptoConfig, byte[])}
```

Comment:
There is value is still having a decrypt method which takes a `ParsedCiphertext` as an argument. That lets a caller inspect the ciphertext prior to decryption without incurring a double parsing cost.

Reply:
Where does a caller have access to a ParsedCiphertext without creating it themselves? The AwsCryptoResult has CiphertextHeaders, but no ParsedCiphertext

Reply:
They would create it themselves. That is allowed. Then they can inspect the headers prior to decryption (though they have to remember that the data is untrusted at that point).

Reply:
Got it, I've made decryptData( ParsedCipherText) public